isobus/src/can_network_manager.cpp

```diff
@@ -608,6 +632,16 @@ namespace isobus
 			}
 			protocolPGNCallbacksMutex.unlock();
 
+			anyControlFunctionCallbacksMutex.lock();
```

Comment:
Good question. I was using lock/unlock here just to keep it locked for the minimum amount of time, and because it's a bit sketchy to lock multiple mutexes at a time, which is the recipe for a deadlock. You are right about the exceptions perhaps being a concern, but as it stands right now we are not handling any exceptions, so it's likely the program would just terminate rather than continue if there is some issue. Though, that assumption may not carry through to a consumer application so perhaps it is still a concern. It may be that a better solution is to move those mutex protected operations into functions and use `std::lock_guard` to lock them? Technically we don't need `unique_lock` because these two PGN callback mutexes aren't used with condition variables and lock_guard would do the trick.

Reply:
Yea, IIRC unique_lock also has a .unlock() function which could be called. Though I'm definitely not knowledged enough to know what the optimal choice would be. So feel free to do what you think is best

Reply:
Updated to move all the content of `process_rx_messages` into functions, and changed those functions to use `lock_guard` for each mutex